pkg/storage/gcp/lister.go

```diff
@@ -0,0 +1,45 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	athensStorage "github.com/gomods/athens/pkg/storage"
+	"google.golang.org/api/iterator"
+)
+
+// List implements the (./pkg/storage).Lister interface
+// It returns a list of versions, if any, for a given module
+func (s *Storage) List(module string) ([]string, error) {
+	ctx := context.Background()
+	// get object iterator for module matches
+	it := s.bucket.Objects(ctx, &storage.Query{Prefix: module})
+
+	versions := make([]string, 0, 10)
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("could not iterate over query: %s", err)
+		}
+
+		// kinda hacky looking at this time
+		if strings.HasSuffix(attrs.Name, ".info") {
+			segments := strings.Split(attrs.Name, "/")
+			// version should be last segment w/ .info suffix
+			last := segments[len(segments)-1]
+			version := strings.TrimSuffix(last, ".info")
+			versions = append(versions, version)
+		}
+	}
```

Comment:
Sorry if I'm missing something completely obvious but shouldn't `List` be just a file which we update after Save and serve from CDN?

Reply:
That feels like a more performant solution, so long as we make it concurrency safe. I mostly replicated functionality I saw the `fs` storage pkg providing.

Reply:
Or is it meant for proxy?

Reply:
It is part of the (./pkg/storage).Backend interface. Lister, Getter, Checker, Saver and Deleter

Reply:
If we distribute the files via CDN then we can download them from url no?

Reply:
But it could be still useful if the storage is used by proxy to store private modules.